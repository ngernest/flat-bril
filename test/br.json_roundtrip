{
  "functions": [
    {
      "args": [],
      "instrs": [
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 4
        },
        {
          "dest": "b",
          "op": "const",
          "type": "bool",
          "value": false
        },
        {
          "args": [
            "b"
          ],
          "funcs": [],
          "labels": [
            "there",
            "here"
          ],
          "op": "br"
        },
        {
          "label": "here"
        },
        {
          "dest": "v",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "label": "there"
        },
        {
          "args": [
            "v"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
