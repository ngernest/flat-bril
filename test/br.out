@main
	op: const		dest: v 		type: int		value: 4	
	op: const		dest: b 		type: bool		value: false	
	op: br   		args: ["b"]		labels: ["there", "here"]
.here
	op: const		dest: v 		type: int		value: 2	
.there
	op: print		args: ["v"]	
